File: Computabilidade-e-Complexidade-de-Algoritmos-main/python/4.py
def afd_ao_menos_um_zero(s):
    estado = 'q0'
    for char in s:
        if estado == 'q0' and char == '0':
            estado = 'q1'
    return estado == 'q1'

# Testes
print(afd_ao_menos_um_zero("110"))  # True
print(afd_ao_menos_um_zero("111"))  # False

File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/4.java
public class AFDAoMenosUmZero {
    public static boolean aoMenosUmZero(String s) {
        String estado = "q0";
        for (char c : s.toCharArray()) {
            if (estado.equals("q0") && c == '0') {
                estado = "q1";
            }
        }
        return estado.equals("q1");
    }

    public static void main(String[] args) {
        System.out.println(aoMenosUmZero("110"));  // True
        System.out.println(aoMenosUmZero("111"));  // False
    }
}
